modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java

```diff
@@ -1103,6 +1110,23 @@ private IgniteCheckedException shutdown(CacheStoreHolder holder, boolean cleanFi
         return aggr;
     }
 
+    private void cleanParentDirectories(CacheStoreHolder holder) {
+        Arrays.stream(holder.partStores).map(store -> ((FilePageStoreV2)store).getPath().getParent()).distinct().forEach(new Consumer<Path>() {
```

Comment:
All files in a store should have the same parent, it should be enough to read it for index partition, for example.
Or, if you want to be maximally flexible, you should use `findAny` instead of `distinct` stream.